### ISSUE TITLE
[css-env-1] ua should be able to update values

### ISSUE THREAD

**Issue by rebeccahughes** [w3c/csswg-drafts#2634]
Spec: https://drafts.csswg.org/css-env-1/

The user agent should be able to update user agent values when needed (e.g. the orientation of the page changes). I am not sure how this would affect substitution.

**Comment by upsuper** [comment]
Reparse on computation, right? I don't think we want to reparse the specified value... It should just behave like `var()`.

**Comment by tabatkins** [comment]
There's an open issue for that. We might want to do it on specified values. There's no concept of "computed value" outside of properties, after all; when would a var()-equivalent substitution happen for `@media (width <= env(foo)) {...}`?

**Comment by upsuper** [comment]
On evaluation time.

Style engines should not be asked to reparse the sheet from CSS in general because the in-memory data may be updated via CSSOM, and you can get new declaration from script, so anything which cannot be preserved in specified value is basically lost forever.

**Comment by tabatkins** [comment]
What do you mean by "evaluation time"? Like, if I use top-notch-size in a MQ, and you rotate the phone so it swaps from zero to non-zero, can the MQ change whether it matches?

**Comment by upsuper** [comment]
Yeah, it should get updated, just like when you resize the viewport / move the window to a monitor with different dpi, etc.
